planner/core/planbuilder.go

```diff
@@ -523,6 +526,9 @@ func (b *PlanBuilder) GetDBTableInfo() []stmtctx.TableEntry {
 
 // GetOptFlag gets the optFlag of the PlanBuilder.
 func (b *PlanBuilder) GetOptFlag() uint64 {
+	if b.isSampling {
```

Comment:
Why do we need to return 0 for sampling?

Reply:
This is used to disable logic optimization.